cmake/define.inc

```diff
@@ -224,9 +224,9 @@ IF (TD_DARWIN_64)
   ADD_DEFINITIONS(-DUSE_LIBICONV)
   MESSAGE(STATUS "darwin64 is defined")
   IF ("${CPUTYPE}" STREQUAL "apple_m1")
-    SET(COMMON_FLAGS "-Wall -Werror -Wno-missing-braces -fPIC -D_FILE_OFFSET_BITS=64 -D_LARGE_FILE")
+    SET(COMMON_FLAGS "-Wall -Werror -Wno-missing-braces -Wno-tautological-constant-out-of-range-compare -Wno-unused-but-set-variable -fPIC -D_FILE_OFFSET_BITS=64 -D_LARGE_FILE")
```

Comment:
To be consistent on compiler warnings among Darwin, Linux, ARM64 and Win OSes at the this point. 
In OSes other than Darwin, the compilers (i.e. GCC in Linux and so) ignores the tautological assert statements which are always true without complaining.

Reply:
In Darwin, Clang's compiler warning on unused-but-set-variable is not accurate. Disable it too.